vortex-serde/src/layouts/read/layouts/column.rs

```diff
@@ -47,98 +53,569 @@ impl LayoutSpec for ColumnLayoutSpec {
 pub struct ColumnLayout {
     fb_bytes: Bytes,
     fb_loc: usize,
+    length: u64,
+    offset: usize,
     scan: Scan,
-    layout_builder: LayoutDeserializer,
+    layout_serde: LayoutDeserializer,
     message_cache: RelativeLayoutCache,
-    reader: Option<BatchReader>,
+    reader: Option<ColumnBatchReader>,
 }
 
 impl ColumnLayout {
     pub fn new(
         fb_bytes: Bytes,
         fb_loc: usize,
+        length: u64,
         scan: Scan,
-        layout_builder: LayoutDeserializer,
+        layout_serde: LayoutDeserializer,
         message_cache: RelativeLayoutCache,
     ) -> Self {
         Self {
             fb_bytes,
             fb_loc,
             scan,
-            layout_builder,
+            length,
+            layout_serde,
             message_cache,
             reader: None,
+            offset: 0,
         }
     }
 
-    pub fn flatbuffer(&self) -> fb::Layout {
+    pub fn flatbuffer(&self) -> footer::Layout {
         unsafe {
             let tab = flatbuffers::Table::new(&self.fb_bytes, self.fb_loc);
-            fb::Layout::init_from_table(tab)
+            footer::Layout::init_from_table(tab)
         }
     }
 
     fn read_child(
         &self,
         idx: usize,
-        children: Vector<ForwardsUOffset<fb::Layout>>,
+        children: Vector<ForwardsUOffset<footer::Layout>>,
         dtype: DType,
     ) -> VortexResult<Box<dyn LayoutReader>> {
-        let layout = children.get(idx);
-
-        // TODO: Figure out complex nested schema projections
-        let mut child_scan = self.scan.clone();
-        child_scan.projection = Projection::All;
-
-        self.layout_builder.read_layout(
+        let mut layout = self.layout_serde.read_layout(
             self.fb_bytes.clone(),
-            layout._tab.loc(),
-            child_scan,
+            children.get(idx)._tab.loc(),
+            self.length,
+            // TODO(robert): Changes this once we support nested projections
+            Scan::new(None),
             self.message_cache.relative(
                 idx as u16,
                 Arc::new(LazyDeserializedDType::from_dtype(dtype)),
             ),
-        )
+        )?;
+        if self.offset != 0 {
+            layout.advance(self.offset)?;
+        }
+        Ok(layout)
     }
 
-    pub fn lazy_dtype(&self) -> VortexResult<Arc<LazyDeserializedDType>> {
-        match &self.scan.projection {
-            Projection::All => Ok(self.message_cache.dtype().clone()),
-            Projection::Flat(p) => self.message_cache.dtype().project(p),
+    fn filter_reader(&mut self) -> VortexResult<ColumnBatchReader> {
```

Comment:
This method and read_children can be unified. Just need to be careful about field ordering